sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/AmqpMessageConverter.cs

```diff
@@ -144,12 +144,20 @@ private static ArraySegment<byte> ReadStreamToArraySegment(Stream stream)
             {
                 case BufferListStream bufferListStream:
                     return bufferListStream.ReadBytes((int)stream.Length);
+
+                case MemoryStream memStreamSource:
+                {
+                    using var memStreamCopy = new MemoryStream((int)(memStreamSource.Length - memStreamSource.Position));
```

Comment:
Interesting, do we know how this block would ever be hit? I thought we always get back a BufferedListStream.

Reply:
Outside of testing, it wouldn't get hit for the current AMQP library implementation.   As you mentioned, that is always sending the `BufferedListStream` back.   That said, it seemed like a reasonable fallback, since the method is fairly genetic and the AMQP implementation may or may not change in the future.